src/main/java/net/atos/zac/app/taken/TakenRESTService.java

```diff
@@ -165,11 +165,13 @@ public RESTTaak readTaak(@PathParam("taskId") final String taskId) {
     @PUT
     @Path("taakdata")
     public RESTTaak updateTaakdata(final RESTTaak restTaak) {
-        final Task task = takenService.readOpenTask(restTaak.id);
+        Task task = takenService.readOpenTask(restTaak.id);
         assertPolicy(getTaakStatus(task) != AFGEROND && policyService.readTaakRechten(task).wijzigen());
         taakVariabelenService.setTaakdata(task, restTaak.taakdata);
         taakVariabelenService.setTaakinformatie(task, restTaak.taakinformatie);
-        updateTaak(restTaak);
+        task = updateTaak(restTaak);
+        eventingService.send(TAAK.updated(task));
```

Comment:
updateTaak is now a private method that only persists data, so the responsibility to send these two events now lies with this endpoint. See comment below.

Reply:
Method `updateTaak` is reloading the `Task` from the `takenService`, and doesn't do much with it. I would incorporate the couple of lines of code here instead.

Reply:
That's fair. It results in a very slight duplication of code between updateTaakdata and completeTaak but I think that's not an issue. Will do, thanks!

Reply:
Done :)